src/endpoint/s3/ops/s3_post_bucket_delete.js

```diff
@@ -23,6 +24,23 @@ async function post_bucket_delete(req) {
     }
 
     dbg.log3('post_bucket_delete: objects', objects);
+    // In this section we will remove duplication from object array
```

Comment:
@shirady Let me suggest a variations of the code, to eliminate the JSON.stringify call which is much heavier than a simple string concat. Using map/set is also a better idea in cases where the uniq keys are long and random.

```js
const delete_list = req.body.Delete?.Object || [];
if (delete_list.length > 1000) {
    throw ...
}

const uniq_map = new Map();
for (const item of delete_list) {
    const key = item.Key?.[0];
    const version_id = item.VersionId?.[0];
    const key_version = (key || '') + '\0' + (version_id || ''); // using null char (\x00) as separator
    uniq_map.set(key_version, { key, version_id });
}
const objects = Array.from(uniq_map.values());
```

Reply:
I changed it by your suggestion, although I didn't understand why you use `req.body.Delete?.Object` and not just `req.body.Delete.Object` since Object is required:

> Object
> The object to delete.
> Type: Array of ObjectIdentifier data types
> Required: Yes

Would you please explain why?
To match the style (and avoid a DeepScan issue) I added the optional chaining operator (`?.`) to `quiet`.

Reply:
`req.body.Delete?.Object` checks if `req.body.Delete` is defined, and only if it is, it goes to `Object`

Reply:
Thanks, but `Delete` is also required as well...
```
Delete
Root level tag for the Delete parameters.
Required: Yes
```
And before this change, we didn't check if it is non-null or non-defined and was fine... (or not?).

Reply:
because the user might not be so dependable and might not send it in the xml...

Reply:
Before the change (in addition to the bug fix) we also didn't parse the XML, so in case the client didn't send the XML with the `Delete` part we would have an error that we cannot read properties of undefined (tried to read 'Quiet').

As discussed, there are a few operations in which the body we get an XML is low (I checked - 22 operations in `s3/ops` we implemented approximately half of them).

There are two options to handle the missing `Delete` part:
1. Ignore it and use an empty array of the deleted list (the suggested change in first commit).
2. Throw an error when we start to parse the XML and don't find a required field.
I was convinced when we talked about the first option, but now that I'm looking at our examples I tend to go by the second option, for example: `put_bucket_tagging(req)` uses a function `parse_body_tagging_xml(req)` and inside it the first check is:
     
```
const tagging = req.body.Tagging;
    if (!tagging) throw new S3Error(S3Error.MalformedXML);
```
* I added the second option in a separate commit.